### ISSUE TITLE
Unable to capture non-Wing CDK constructs

### ISSUE THREAD

**Issue by eladb** [winglang/wing#1878]
### Community Note

> Please vote by adding a 👍 reaction to the issue to help us prioritize.
> If you are interested to work on this issue, please leave a comment.

### I tried this:

```js
bring cloud;
bring "@cdktf/provider-aws" as tfaws;

class Table {
  table: tfaws.dynamodbTable.DynamodbTable;

  init(table_name: str, hash_key: str) {
    this.table = new tfaws.dynamodbTable.DynamodbTable(
      name: table_name,
      billing_mode: "PAY_PER_REQUEST",
      hash_key: hash_key,
      attribute: [{ name: hash_key, type: "S" }]
    );
  }

  inflight print_name() { 
    log(this.table.name);
  }
}

let table = new Table("my-first-table", "id");

new cloud.Function(inflight () => {
  table.print_name();
});
```

### I expected this:

To print the table name in inflight

### Instead, this happened:

```
preflight error: unable to serialize immutable data object of type DynamodbTable

  note: intermediate javascript code (/Users/eladb/playground/dynamodb/target/main.tfaws/.wing/preflight.js):

     	_toInflight() {
>>   	const table_client = this._lift(this.table);
     	const self_client_path = "./clients/Table.inflight.js".replace(/\\/g, "/");
     	return $stdlib.core.NodeJsCode.fromInline(`(new (require("${self_client_path}")).Table({table: ${table_client}}))`);
--------------------------------- STACK TRACE ---------------------------------
/Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/core/internal.js:89
    throw new Error(`unable to serialize immutable data object of type ${obj.constructor?.name}`);
    ^

Error: unable to serialize immutable data object of type DynamodbTable
    at serializeImmutableData (/Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/core/internal.js:89:11)
    at Table._lift (/Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/core/resource.js:250:54)
    at Table._toInflight (evalmachine.<anonymous>:36:30)
    at serializeImmutableData (/Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/core/internal.js:76:28)
    at makeHandler (/Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/core/internal.js:10:22)
    at new Inflight (/Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/core/inflight.js:74:43)
    at new MyApp (evalmachine.<anonymous>:44:73)
    at evalmachine.<anonymous>:55:1
    at Script.runInContext (node:vm:141:12)
    at Object.runInContext (node:vm:297:6)
    at compile (/Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/dist/commands/compile.js:172:12)
```

### Component:

SDK

### Anything else?

We currently only support Wing resources in the capture mechanism. We need to support arbitrary constructs (`Construct.isConstruct(x)`). In those cases, there is no `_toInflight()` method of course.

Under `internal.js`:

```
// if the object is a resource (i.e. has a "_toInflight" method"), we use it to serialize
// itself.
if (typeof obj._toInflight === "function") {
    return obj._toInflight().text;
}
```

**Comment by skorfmann** [comment]
Just came across this issue when trying to test cdktf resources.

```
test "some cdktf resource" {
  assert(resource.foo == "bar");
}
```

Would be great if that was working

**Comment by eladb** [comment]
Working on it...

**Comment by staycoolcall911** [comment]
Closed in #3213
